app/assets/javascripts/hyrax/collections.js

```diff
@@ -240,27 +240,29 @@ Blacklight.onLoad(function () {
     };
     var $deleteWordingTarget = $('#selected-collections-delete-modal .pluralized');
 
-    tableRows.each(function(i, row) {
-      checkbox = $(row).find('td:first input[type=checkbox]');
-      if (typeof checkbox[0] !== "undefined") {
-        if (checkbox[0].checked) {
-          numRowsSelected++;
-        }
-      }
-    });
+    var canDeleteAll = true;
+    var selectedInputs = $('#documents table.collections-list-table tbody tr')
+      // Get all inputs in the table
+      .find('td:first input[type=checkbox]')
+      // Filter to those that are checked
+      .filter((i, checkbox) => checkbox.checked)
+
+    var cannotDeleteInputs = selectedInputs.filter((i, checkbox) => (checkbox.dataset.hasaccess === "false"))
+    if(cannotDeleteInputs.length > 0) {
+      // TODO: Can we pass data to this modal to be more specific about which ones they cannot delete?
```

Comment:
Regarding this "TODO", since we're re-using one modal partial and the content is being built dynamically, probably the best way to specify data you want to access in the modal is to add it to the table row, or table, or wherever makes sense, `data-` attributes.  Maybe as a stringify'd array or something?